File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rrl(self,node,level,res):
        if node is None:
            return 
        if level==len(res):
            res.append(node.val)
        self.rrl(node.right,level+1,res)
        self.rrl(node.left,level+1,res)    

    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res=[]
        level=0
        self.rrl(root,level,res)
        return res
        

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int>res;
        int level=0;
        rrl(root,level,res);
        return res;
    }
    void rrl(TreeNode* node,int level,vector<int>& res){
        if(node==NULL)return;
        if(level==res.size())res.push_back(node->val);
        rrl(node->right,level+1,res);
        rrl(node->left,level+1,res);
    }
};